codelab/JinghangCodeLab/MagicNumberConverter.java

```diff
@@ -0,0 +1,41 @@
+package JinghangCodeLab;
+
+public class MagicNumberConverter {
+
+    public static int binaryToDecimal(String binary)
+    {  
+        //edge cases
+        if (binary == "0"){
+            return 0;
+        }
+
+        //negative
+        int isNegative = -1;
+        if(Integer.parseInt(binary) < 0){
+            isNegative = 0;
+        }
+        else{
+            isNegative = -1;
+        }
+        //formula
+        int length =  binary.length() - 1;
+        int multiple = 1;
+        int sum = 0;
+        for (int i = length; i > isNegative; i--){
```

Comment:
i > isNegative is a little bit confusing and not easy to understand. Try to create another variable to keep track of the index of the digits.